application/backend/server/src/paste/manager.rs

```diff
@@ -0,0 +1,133 @@
+use crate::app::state::AppState;
+use crate::common::ResponseInfo;
+use crate::common::{session::validate_token, InternalServerError, PasteData};
+use actix_web::http::header;
+use actix_web::{delete, get, post, web, HttpRequest, HttpResponse, Responder};
+use log::info;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize)]
+struct AddPasteReq {
+    text: String,
+}
+
+#[derive(Serialize)]
+struct GetPasteResp {
+    text: String,
+}
+
+#[post("/paste")]
+pub async fn add_paste(
+    state: web::Data<AppState>,
+    add_paste_req: web::Json<AddPasteReq>,
+    req: HttpRequest,
+) -> impl Responder {
+    info!("Inserting paste.");
+
+    let result = save_paste(&state, &add_paste_req, &req).await;
+    match result {
+        Ok(_) => {
+            info!("Paste added successfully.");
+            HttpResponse::Created().json({
+                ResponseInfo {
+                    info: "Paste added successfully.".to_string(),
+                }
+            })
+        }
+        Err(e) => e.handle_error_for_http_resp(),
+    }
+}
+
+async fn save_paste(
+    state: &web::Data<AppState>,
+    add_paste_req: &web::Json<AddPasteReq>,
+    req: &HttpRequest,
+) -> Result<(), InternalServerError> {
+    check_session(&req)?;
+
+    let client_resp = state.get_key_client().get_key().await?;
+    let paste_data = PasteData {
```

Comment:
how is paste author associated with their paste?

Reply:
For now paste owner is not established. It will be introduced after requirements finalization, when all user scenarios will be specified.